File: dfs.py
def dfs_recursive(graph, node, visited):
    visited[node] = True
    print(node, end=" ")

    for neighbor in graph[node]:
        if not visited[neighbor]:
            dfs_recursive(graph, neighbor, visited)

# Example
graph = {
    0: [1, 2],
    1: [0, 3, 4],
    2: [0, 5],
    3: [1],
    4: [1],
    5: [2]
}

visited = [False] * len(graph)
print("DFS Recursive:", end=" ")
dfs_recursive(graph, 0, visited)

File: dfs.cpp
#include <iostream>
#include <vector>
using namespace std;

void dfs_recursive(int node, vector<vector<int>>& graph, vector<bool>& visited) {
    visited[node] = true;
    cout << node << " ";

    for (int neighbor : graph[node]) {
        if (!visited[neighbor]) {
            dfs_recursive(neighbor, graph, visited);
        }
    }
}

int main() {
    int n = 6; 
    vector<vector<int>> graph(n);

    graph[0] = {1, 2};
    graph[1] = {0, 3, 4};
    graph[2] = {0, 5};
    graph[3] = {1};
    graph[4] = {1};
    graph[5] = {2};

    vector<bool> visited(n, false);

    cout << "DFS Recursive: ";
    dfs_recursive(0, graph, visited);

    return 0;
}
